presto-main/src/main/java/com/facebook/presto/metadata/Metadata.java

```diff
@@ -73,6 +73,11 @@
 
     List<String> listSchemaNames(Session session, String catalogName);
 
+    default Optional<List<String>> getPartitionNames(Session session, String databaseName, TableHandle tableHandle)
```

Comment:
I think we have a way to hide this within Hive connector. Will add this point to design doc.